packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -3452,6 +3452,42 @@ void main() {
     textFieldPosition = tester.getTopLeft(find.byType(TextField));
     expect(textFieldPosition, equals(const Offset(16.0, 544.0)));
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/143505.
+  testWidgets('When using the keyboard, ensure that the focus is on the DropdownMenu', (WidgetTester tester) async {
+    final FocusNode focusNode = FocusNode();
+    addTearDown(focusNode.dispose);
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Center(
+            child: DropdownMenu<TestMenu>(
+              focusNode: focusNode,
+              dropdownMenuEntries: menuChildren,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+    await tester.pump();
+
+    await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+    await tester.pump();
+
+    await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+    await tester.pump();
+    /*Now the focus is on the icon button.*/
+
+    await tester.sendKeyEvent(LogicalKeyboardKey.enter);
+    await tester.pump();
+
+    await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
+    await tester.pump();
+
+    expect(focusNode.hasPrimaryFocus, true);
+  }, variant:TargetPlatformVariant.desktop());
```

Comment:
```suggestion
  });
```

Keyboard navigation is also possible on mobile.
After removing this variant, set `requestFocusOnTap` to true on the DropdownMenu (line 3466), this way the initial focus state will be the same for mobile and desktop.
See https://github.com/flutter/flutter/pull/156131 where the desktop variant was removed for several tests.